docs/notebooks/shard_map.ipynb

```diff
@@ -827,17 +827,20 @@
     "Specs = PyTree[PartitionSpec]\n",
     "\n",
     "def shard_map(\n",
-    "    f: Callable, mesh: Mesh, in_specs: Specs, out_specs: Specs,\n",
-    "    auto: collections.abc.Set[AxisName] = frozenset([]),\n",
+    "    f: Callable, /, *, out_specs: Specs, mesh: Mesh | None = None,\n",
+    "    in_specs: Specs | None = None,\n",
+    "    axis_names: collections.abc.Set[AxisName] = set(),\n",
     "    check_vma: bool = True,\n",
     ") -> Callable:\n",
     "  ...\n",
     "```\n",
     "where:\n",
     "* communication collectives like `psum` in the body of `f` can mention the axis names of `mesh`;\n",
-    "* `mesh` encodes devices arranged in an array and with associated axis names, just like it does for `sharding.NamedSharding`;\n",
-    "* `in_specs` and `out_specs` are `PartitionSpec`s which can affinely mention axis names from `mesh` to express slicing/unconcatenation and concatenation of inputs and outputs, respectively, with unmentioned names corresponding to replication and untiling (assert-replicated-so-give-me-one-copy), respectively;\n",
-    "* `auto` is an optional set of axis names corresponding to the subset of names of `mesh` to treat automatically in the body, as in the caller, rather than manually;\n",
+    "* `mesh` encodes devices arranged in an array and with associated axis names, just like it does for `sharding.NamedSharding`; If None, mesh will be inferred from the\n",
+    "context which can be set via `jax.sharding.use_mesh` context manager.\n",
+    "* `in_specs` are `PartitionSpec`s which can affinely mention axis names from `mesh` to express slicing/unconcatenation of inputs, respectively, with unmentioned names corresponding to replication and untiling (assert-replicated-so-give-me-one-copy). If None, all mesh axes must be of type `Explicit`, in which case the in_specs are inferred from the argument types;\n",
```

Comment:
"affinely" is such a pretentious word to use here lol, sorry

you don't have to fix it here but we should say "zero or one times" instead

Reply:
Done